js/app/demo/editor.jsx

```diff
@@ -50,10 +50,24 @@ define(["react", "orion", "reactDom", "events"], function(React, orion, ReactDOM
         componentWillReceiveProps: function(nextProps) {
             if (nextProps.errors) {
                 this.editor.showProblems(nextProps.errors.map(function(error) {
+                    if (error.fatal) {
+                        return {
+                            line: error.line,
+                            start: error.column,
+                            end: error.column + 1,
+                            description: error.message,
+                            severity: "error"
+                        };
+                    }
+                    var start = nextProps.getIndexFromLoc({ line: error.line, column: error.column - 1 });
+                    var end = nextProps.getIndexFromLoc({
+                        line: error.endLine || error.line,
+                        column: (error.endColumn || error.column) - 1
+                    });
+
                     return {
-                        line: error.line,
-                        start: error.column,
-                        end: error.column + 1,
+                        start: start,
+                        end: start === end ? end + 1 : end,
```

Comment:
Is there documentation for the shape of the objects passed to `showProblems` somewhere? The [docs I found](https://orionhub.org/jsdoc/symbols/orion.editor.Editor.html#showProblems) just say `showProblems(problems)`, which isn't very helpful. I'm only asking because the fatal error condition includes a `line` property and this doesn't, and I just wanted to make sure that's intentional.

Reply:
I couldn't find any, unfortunately -- I discovered this API by looking at [the source code](https://github.com/eclipse/orion.client/blob/ee564a2964ca2ead6cfe6752c13d50abf8b86c50/bundles/org.eclipse.orion.client.editor/web/orion/editor/editor.js#L1241-L1262). It appears that when `line` is provided, `start` and `end` are considered to be column offsets, otherwise `start` and `end` are considered to be global indices in the text.